src/cart/utils/crt_utils.c

```diff
@@ -346,12 +349,141 @@ crtu_load_group_from_file(const char *grp_cfg_file, crt_context_t ctx,
 	return rc;
 }
 
+#define SYS_INFO_BUF_SIZE 16
+
+int
+crtu_dc_mgmt_net_cfg_rank_add(const char *name, crt_group_t *group,
+		    crt_context_t *context)
+{
+	int	i;
+	int rc = 0;
+	struct dc_mgmt_sys_info crt_net_cfg_info;
+	Mgmt__GetAttachInfoResp *crt_net_cfg_resp;
+
+	/* Query the agent for the CaRT network configuration parameters */
+	rc = get_attach_info(name, true /* all_ranks */, &crt_net_cfg_info, &crt_net_cfg_resp);
+	D_ASSERTF(rc == 0, "get_attach_info failed, rc=%d\n", rc);
+
+	for (i = 0; i < crt_net_cfg_resp->n_rank_uris; i++) {
+		Mgmt__GetAttachInfoResp__RankUri *rank_uri = crt_net_cfg_resp->rank_uris[i];
+
+		/* Init eq ref count to 1 */
+		daos_eq_ref_set(1);
+
+		rc = crt_group_primary_rank_add(context, group,
+						rank_uri->rank, rank_uri->uri);
+
+		if (rc != 0) {
+			D_ERROR("failed to add rank %u URI %s to group %s: "
+				DF_RC"\n", rank_uri->rank, rank_uri->uri, name,
+				DP_RC(rc));
+			goto err_group;
+		}
+	}
+
+err_group:
+	if (rc != 0) {
+		crt_group_view_destroy(group);
+	}
+
```

Comment:
since you are not preserving attach info across calls, should this not do put attac info at the end?

Reply:
When I try preserving attach info across calls, I hit this error:
```
[stderr] [1,0]<stderr>:self_test: src/mgmt/svc.pb-c.c:469: mgmt__get_attach_info_resp__free_unpacked: Assertion `message->base.descriptor == &mgmt__get_attach_info_resp__descriptor' failed.
```